srcpkgs/julia/template

```diff
@@ -2,66 +2,64 @@
 pkgname=julia
 version=1.5.0
 revision=1
-archs="i686* x86_64* armv7l aarch64"
+archs="i686* x86_64* armv7l* aarch64*"
 build_style=gnu-makefile
 make_build_args="prefix=/usr sysconfdir=/etc datarootdir=/usr/share
- USE_LLVM_SHLIB=1 USE_BINARYBUILDER=0
- USE_SYSTEM_LIBUV=0 USE_SYSTEM_LIBUNWIND=0 USE_SYSTEM_PATCHELF=1 USE_SYSTEM_LIBM=0
- USE_SYSTEM_DSFMT=0 USE_SYSTEM_LLVM=0 USE_SYSTEM_PCRE=1 USE_SYSTEM_BLAS=0
- USE_SYSTEM_GMP=1 USE_SYSTEM_LIBGIT2=1 USE_SYSTEM_MBEDTLS=1 USE_SYSTEM_LIBSSH2=1
- USE_SYSTEM_CURL=1 USE_SYSTEM_MPFR=1 USE_SYSTEM_SUITESPARSE=0 USE_SYSTEM_UTF8PROC=0
- USE_SYSTEM_ZLIB=1 USE_SYSTEM_P7ZIP=1 USE_SYSTEM_LAPACK=0"
+ USE_BINARYBUILDER=0 USE_SYSTEM_LIBM=0 USE_SYSTEM_DSFMT=0
+ USE_SYSTEM_LIBUV=0 USE_SYSTEM_SUITESPARSE=0
+ USE_SYSTEM_LLVM=1 USE_SYSTEM_LIBUNWIND=1 USE_SYSTEM_PCRE=1
+ USE_SYSTEM_GMP=1 USE_SYSTEM_PATCHELF=1 USE_SYSTEM_MPFR=1
+ USE_SYSTEM_MBEDTLS=1 USE_SYSTEM_LIBSSH2=1 USE_SYSTEM_CURL=1
+ USE_SYSTEM_ZLIB=1 USE_SYSTEM_P7ZIP=1 USE_SYSTEM_LIBGIT2=1
+ USE_SYSTEM_BLAS=1 USE_SYSTEM_LAPACK=1 USE_SYSTEM_UTF8PROC=1
```

Comment:
We either need to let julia vendor in its own openblas or provide a system [openblas it can link against using 64-bit integers (ILP64)](https://github.com/JuliaLang/julia/blob/master/doc/build/distributing.md#notes-on-blas-and-lapack) (this change was made in https://github.com/void-linux/void-packages/pull/20353 because otherwise some 3rd-party julia packages which expect ILP64 don't work, e.g. Arpack.jl). Long term, it would be good to provide this as a system library [like Fedora does](https://discourse.julialang.org/t/no-documentation-for-the-state-of-patches-for-dependencies-of-julia/44735/3), but I haven't had time to work on that change yet.

Reply:
These spurious dependencies aren't really spurious. For some sensless reason, Julia doesn't fix its build system and works around some library issues by directly **copying the system libraries into** `usr/lib/julia` and modifying the rpaths.

Even if we have to build openblas as part of Julia to support ILP64 (which never should have been the case---I don't know of a non-dead architeture that is natively ILP64---why would Julia impose this???), we should just *delete* the "vendored" clones it's made in `/usr/lib/julia` and, if we really need them, symlink to the system libraries.

Reply:
Looking at [how Fedora builds openblas with ILP64 (`libopenblas64_.so`)](https://src.fedoraproject.org/rpms/openblas/blob/5641975e5cba160e947c65c918c30f44b7bc17a5/f/openblas.spec#_411), getting it working on Void might not be so bad. Can take a look when I get a chance. I'd prefer not to block packaging julia 1.5 on this though.

Reply:
>Even if we have to build openblas as part of Julia to support ILP64 (which never should have been the case---I don't know of a non-dead architeture that is natively ILP64---why would Julia impose this???), we should just delete the "vendored" clones it's made in /usr/lib/julia and, if we really need them, symlink to the system libraries.

I'll try this locally and see how it goes.

Reply:
>(which never should have been the case---I don't know of a non-dead architeture that is natively ILP64---why would Julia impose this???)

I went digging: https://github.com/JuliaLang/julia/issues/1527

Looks like blas uses `int` for dimensions, so if you want to have arrays with dimensions greater than 2147483647 you need ILP64. My understanding here is fuzzy at best, so I could be off-base.

Reply:
I'm test-building a version with vendored llvm and openblas that removes the conflicting `libgfortran`, `libgcc_s` and `libquadmath` without creating symlinks in `/usr/lib/julia`. The linker should be able to find these libraries in their default locations. When it's built, I'll force push to your branch (I've rebased on master so we don't have to compile an outdated `libressl`) and hopefully you can run it through its paces.